.github/workflows/pull-request-checks.yaml

```diff
@@ -107,6 +107,7 @@ jobs:
       - name: Run unit tests
         run: |
           make -C unit      test
+          make -C unit      test "[z3]"
```

Comment:
I'm curious why this is in some of the builds but not others. On the one hand I understand that testing this is largely unrelated to the various flavours of make/ubuntu and so testing once per OS is sufficient. On the other hand, this makes the tests between OS (versions) less consistent and maybe has potential for lack of clarity.